user_interactions.rb

```diff
@@ -61,4 +65,17 @@ def list_numbers
     puts list
     gets.chomp
   end
+
+  def exit
+    create_storage_directory unless storage_directory_check?
+    app_arrays = %w[people books rentals]
+
+    app_arrays.each do |file|
+      data = @app.instance_variable_get("@#{file}")
+      json_data = data.map(&:to_hash)
+      create_file(file, json_data)
+    end
+
+    puts 'Thank you for using this app!'
+  end
```

Comment:
- [ ] There seems to be an error somewhere in your app. When I added books, people, and rentals and quit the app, it did not record the new data in the json storage files. Kindly fix that.

- [ ] When I deleted the existing json storage files and created new people, books, and rentals, and quit the app, the files are not created and the data is not saved. The app mostly runs like the previous basic UI project, no new data is persisted in any scenario. Kindly fix these issues.